website/addons/base/__init__.py

```diff
@@ -602,6 +602,8 @@ def get_attached_nodes(self, external_account):
         for node in self.get_nodes_with_oauth_grants(external_account):
             if node is None:
                 continue
+            if node.is_deleted:
```

Comment:
Yes, I think that would make more sense; that way, we don't need to repeat this check if we call `get_nodes_with_oauth_grants` elsewhere.